docs/web3.eth.rst

```diff
@@ -1238,6 +1238,11 @@ Contracts
 
 .. py:method:: Eth.contract(address=None, contract_name=None, ContractFactoryClass=Contract, **contract_factory_kwargs)
 
+    An ``abi`` specifies how a contract can be interacted with. The ``abi`` 
```

Comment:
I feel like this clarification should go down below, as in the "following arguments are accepted" section. This first section discusses the actual named arguments associated with the function, in order. `abi`, while necessary for making contract calls, isn't a named arg, just an optional kwarg (whether this is a good design is perhaps another question :thinking: ) Maybe adding it as a description to the optional kwargs list is a better place? That then opens it as a place for describing usage of the others in that list. 

Just my feels, open for discussion!

Reply:
Would be ideal to call something out here in the contract docs. I wonder if it would help to include abi as an argument in the method signature above, or some way of indicating it is necessary for the interface to work.

Agree this paragraph doesnt seem quite right in the contract API docs. I think it might clutter the list down below, but there is a link to the Contracts guide from there. I think the paragraph might be a good replacement for the contract docs under `Contract.abi`. What do you think?

Reply:
Agree with putting more info under `Contract.abi`. Could even explain what ABI stands for there. I also agree with some indication here that abi is required if you want to interact with the contract.